libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java

```diff
@@ -492,34 +490,35 @@ public boolean removeExternalBlobsDirectory(String soupTableName) {
 	}
 
 	/**
-	 * Re-encrypts the files on external storage with the new key. If external storage is not enabled for any table in the db, this operation is ignored.
+	 * Re-encrypts the files on external storage with the new key. If external storage is not
+	 * enabled for any table in the db, this operation is ignored.
 	 *
-	 * @param db DB containing external storage (if applicable).
+	 * @param db DB containing external storage (if applicable).O
 	 * @param oldKey Old key with which to decrypt the existing data.
 	 * @param newKey New key with which to encrypt the existing data.
 	 */
 	public static void reEncryptAllFiles(SQLiteDatabase db, String oldKey, String newKey) {
```

Comment:
What if this method is called to change key (both using the new encryption scheme)?
Instead of calling Encryptor.legacyDecrypt on line 519, shouldn't you decide based on the key length or something like that?

Reply:
Do we want to support that? I can add that in. However, it would be nice to keep this internal, so apps don't call it incorrectly - goes with the theme of keeping all encryption-related functionality internally handled by the SDK.

Reply:
It's called from the public changeKey method so it should handle any key change (with and without encryption scheme change). If it's only for upgrade purposes, it should have a different name and be called from a upgrade only change key method.
If we can recognize the encryption change by inspecting the key we can have the method handle all cases.

Reply:
Makes sense. Let me look into it.